aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -471,6 +471,121 @@ Tensor _cholesky_helper_cuda_cusolver(const Tensor& self, bool upper) {
   return self_working_copy;
 }
 
+
+template<typename scalar_t>
+inline static void apply_cholesky_cusolver_potrs(Tensor& self_working_copy, const Tensor& A_column_major_copy, bool upper, Tensor& infos) {
+  auto handle = at::cuda::getCurrentCUDASolverDnHandle();
+  const auto uplo = upper ? CUBLAS_FILL_MODE_UPPER : CUBLAS_FILL_MODE_LOWER;
+  const int64_t n = self_working_copy.size(-2);
+  const int64_t nrhs = self_working_copy.size(-1);
+  const int64_t lda = std::max<int64_t>(1, n);
+  const int64_t batch_size = batchCount(self_working_copy);
+  const int64_t self_matrix_stride = matrixStride(self_working_copy);
+  scalar_t* self_working_copy_ptr = self_working_copy.data_ptr<scalar_t>();
+
+  const scalar_t* A_ptr = A_column_major_copy.data_ptr<scalar_t>();
+  const int64_t A_matrix_stride = matrixStride(A_column_major_copy);
+  const int64_t ldb = std::max<int64_t>(1, A_column_major_copy.size(-1));
+
+  int* infos_ptr = infos.data_ptr<int>();
+
+#ifdef USE_CUSOLVER_64_BIT
+  cusolverDnParams_t params;
+  cudaDataType datatype = at::cuda::solver::get_cusolver_datatype<scalar_t>();
+  TORCH_CUSOLVER_CHECK(cusolverDnCreateParams(&params));
+
+  for (int64_t i = 0; i < batch_size; i++) {
+    at::cuda::solver::xpotrs(
+      handle, params, uplo, n, nrhs, datatype,
+      A_ptr + i * A_matrix_stride,
+      lda, datatype,
+      self_working_copy_ptr + i * self_matrix_stride,
+      ldb,
+      infos_ptr
+    );
+  }
+
+  TORCH_CUSOLVER_CHECK(cusolverDnDestroyParams(params));
+#else // USE_CUSOLVER_64_BIT
+  int n_32 = cuda_int_cast(n, "n");
+  int nrhs_32 = cuda_int_cast(nrhs, "nrhs");
+  int lda_32 = cuda_int_cast(lda, "lda");
+  int ldb_32 = cuda_int_cast(ldb, "ldb");
+
+  for (int64_t i = 0; i < batch_size; i++) {
+    at::cuda::solver::potrs<scalar_t>(
+      handle, uplo, n_32, nrhs_32,
+      A_ptr + i * A_matrix_stride,
+      lda_32,
+      self_working_copy_ptr + i * self_matrix_stride,
+      ldb_32,
+      infos_ptr
+    );
+  }
+#endif // USE_CUSOLVER_64_BIT
+}
+
+
+// cusolverDn<t>potrsBatched only supports nrhs == 1
+template<typename scalar_t>
+inline static void apply_cholesky_cusolver_potrsBatched(Tensor& self_working_copy, const Tensor& A_column_major_copy, bool upper, Tensor& infos) {
```

Comment:
I appreciate this being added but we should add a comment that it's unused, why it's here but unused, and link the issue. Otherwise someone might get confused and accidentally delete this as dead code.

Reply:
It is unused when MAGMA is linked, which is >99% case. I'll add a comment to clarify that.